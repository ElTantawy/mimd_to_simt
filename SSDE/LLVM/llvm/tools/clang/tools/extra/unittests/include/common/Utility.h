//=-- clang-modernize/Utility.h - Utility functions and macros---*- C++ -*-===//
//
//                     The LLVM Compiler Infrastructure
//
// This file is distributed under the University of Illinois Open Source
// License. See LICENSE.TXT for details.
//
//===----------------------------------------------------------------------===//

#ifndef CLANG_MODERNIZE_UNITTESTS_UTILITY_H
#define CLANG_MODERNIZE_UNITTESTS_UTILITY_H

// FIXME: copied from unittests/Support/Path.cpp
#define ASSERT_NO_ERROR(x)                                                     \
  if (std::error_code ASSERT_NO_ERROR_ec = x) {                                \
    llvm::SmallString<128> MessageStorage;                                     \
    llvm::raw_svector_ostream Message(MessageStorage);                         \
    Message << #x ": did not return errc::success.\n"                          \
            << "error number: " << ASSERT_NO_ERROR_ec.value() << "\n"          \
            << "error message: " << ASSERT_NO_ERROR_ec.message() << "\n";      \
    GTEST_FATAL_FAILURE_(MessageStorage.c_str());                              \
  } else {                                                                     \
  }

#endif // CLANG_MODERNIZE_UNITTESTS_UTILITY_H
