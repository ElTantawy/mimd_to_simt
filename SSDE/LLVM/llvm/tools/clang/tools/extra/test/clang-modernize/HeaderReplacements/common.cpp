// This is just a dummy run command to keep lit happy. Tests for this file are
// in main.cpp
// RUN: true

#include "common.h"

void func1(int &I) {
}

void func2() {
  container C1;
  container C2;
  for (container::iterator I = C1.begin(), E = C1.end(); I != E; ++I) {
    C2.push_back(*I);
  }
}
