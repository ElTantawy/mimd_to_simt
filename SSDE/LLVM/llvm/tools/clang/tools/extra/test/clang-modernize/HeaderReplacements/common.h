#ifndef CPP11_MIGRATE_TEST_HEADER_REPLACEMENTS_COMMON_H
#define CPP11_MIGRATE_TEST_HEADER_REPLACEMENTS_COMMON_H

struct container {
  struct iterator {
    int &operator*();
    const int &operator*() const;
    iterator &operator++();
    bool operator!=(const iterator &other);
  };

  iterator begin();
  iterator end();
  void push_back(const int &);
};

void func1(int &I);
void func2();

void dostuff() {
  container C;
  for (container::iterator I = C.begin(), E = C.end(); I != E; ++I) {
    func1(*I);
  }
}

#endif // CPP11_MIGRATE_TEST_HEADER_REPLACEMENTS_COMMON_H
