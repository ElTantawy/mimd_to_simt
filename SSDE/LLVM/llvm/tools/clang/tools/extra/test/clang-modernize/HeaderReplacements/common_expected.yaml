---
MainSourceFile:  '$(path)/common.cpp'
Replacements:    
  - FilePath:        '$(path)/common.h'
    Offset:          506
    Length:          2
    ReplacementText: elem
  - FilePath:        '$(path)/common.h'
    Offset:          432
    Length:          61
    ReplacementText: '(auto & elem : C)'
  - FilePath:        '$(path)/common.cpp'
    Offset:          289
    Length:          2
    ReplacementText: elem
  - FilePath:        '$(path)/common.cpp'
    Offset:          206
    Length:          63
    ReplacementText: '(auto & elem : C1)'
...
